README.md

```diff
@@ -1,3 +1,134 @@
-MozDef event framework PoC using AWS + Serverless framework
+# MozDef Event Framework
+
+A PoC using AWS + Serverless framework
 
 This is a work in progess which will eventually be used as a template for external event sources.
+
+The idea behind this work was to create a repeatable process for introducing new event sources to MozDef without having to specially craft it every time. While this is new and we haven't onboarded many sources yet, it is expected the function library may grow, and with it we will introduce further changes to this framework as it becomes more mature. 
+
+For example, being able to handle SSO, JWT, or other methods of authorizing event sources are in our interest to support, as well as the ability to reach out and pull down data from Rest API's using those methods of authorization. With the introduction of this framework we should be able to have a repeatable CI/CD pipeline, and be able to add to the library many functions to handle various types of connectivity and data retrieval without much additional work.
+___
+## Framework Components:
+
+__Cloudformation:__ Cloudformation will create the CI/CD components utilizing:
+- __CodePipeline__
+- __CodeCommit__
+- __CodeBuild__
+- __s3__
+- __CloudWatch__
+- __GitHub__.
+
+The intended behavior is to pull the framework from GitHub, and the custom configuration for a specific event source from its CodeCommit repository. The Cloudformation template has the potential to be separately deployed for various event sources, while the only difference will be the event source's CodeCommit repository that contains the custom configuration data for that event source. This allows us to maintain a generic framework that is event source agnostic and relieved of sensitive data. This structure presents the ability to maintain secret data, and preserve it's confidentiality.
+
+__Serverless:__ The Serverless framework is utilized to deploy the actual application that will handle the event source data. This framework creates the following resources:
+ - A custom authorizer for webhook based APIs that use a header token.
```

Comment:
Consider replacing "a header token" with "an Authorization header".

Reply:
Thanks, that does sound better

Reply:
Added!